mongo/client.go

```diff
@@ -294,6 +311,50 @@ func (c *Client) connect() error {
 	return nil
 }
 
+// AppendDriverInfo appends the provided DriverInfo to the driver information
+// that will be sent to the server in handshake requests when establishing new
+// connections. The provided info will overwrite any existing values.
+//
+// AppendsDriverInfo appends the provided [options.DriverInfo] to the metadata
+// (e.g. name, version, platform) that will be sent to the server in handshake
+// requests when establishing new connections. The provided info will overwrite
+// any existing values.
+//
+// Repeated calls to appendMetadata with equivalent DriverInfo is a no-op.
+//
+// Metadata is limited to 512 bytes; any excess will be truncated.
+func (c *Client) AppendDriverInfo(info options.DriverInfo) {
```

Comment:
Instead of a method on `Client`, can we make this an internal option on `Collection`?

E.g.
```go
func blah(coll *mongo.Collection, ...) {
    opts := options.Collection()
    info := options.DriverInfo{
        // New driver info fields.
    }
    xoptions.SetInternalCollectionOptions(opts, "driverInfo", info)
    coll = coll.Clone(opts)
    // Use new collection.
```

Reply:
After attempting to implement the DriverInfo update as an experimental option on Collection, I think the existing API on Client is clearer and simpler. Additionally, if we're going to use the function in 3rd party libraries, we're going to have to support it for the entire v2 major version either way, so making it an internal-only option doesn't align with that use case. Consider this discussion resolved.